pkg/security/secl/compiler/eval/variables.go

```diff
@@ -388,14 +393,23 @@ func (m *MutableIntArrayVariable) Append(_ *Context, value interface{}) error {
 func (m *MutableIntArrayVariable) GetEvaluator() interface{} {
 	return &IntArrayEvaluator{
 		EvalFnc: func(*Context) []int {
-			return m.Values
+			return m.LRU.Keys()
 		},
 	}
 }
 
 // NewMutableIntArrayVariable returns a new mutable integer array variable
-func NewMutableIntArrayVariable() *MutableIntArrayVariable {
-	return &MutableIntArrayVariable{}
+func NewMutableIntArrayVariable(size int, ttl time.Duration) *MutableIntArrayVariable {
+	if size == 0 {
+		size = defaultMaxVariables
+	}
+
+	lru := ttlcache.New(ttlcache.WithCapacity[int, bool](uint64(size)), ttlcache.WithTTL[int, bool](ttl))
+	go lru.Start()
```

Comment:
Where is this stopped ? Otherwise you are leaking a goroutine

Reply:
This is stopped in RuleSet.Release